### PR TITLE
feat: add GitHub logo and link to the website repo

### LINKED ISSUE #285: 📝 Documentation: GitHub Icon on Homepage
### Bug Report Checklist

- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/boston-ts-website/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

GitHub icon would be awesome linking to this repo on the website.

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by DannyGarciaDEV**
## PR Checklist

- [x] Addresses an existing open issue: fixes #285
- [x] That issue was marked as [`Documentation: GitHub Icon on Homepage #285`](https://github.com/SquiggleTools/boston-ts-website/issues/285)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/boston-ts-website/blob/main/.github/CONTRIBUTING.md) were taken

### Description

Added GitHub icon to website header for enhanced navigation.

### Changes Made

- Created `GitHubLogo.tsx` component to render GitHub icon.
- Updated `SocialsList.tsx` to include GitHubLogo component for displaying GitHub link in the socials list.

### How it Works

The new GitHub icon now appears in the website header, linking users directly to the project's GitHub repository.
CodeTS

### Related Issues

- Addresses #285  (GitHub integration task)

### CODE REVIEW

File: app/components/SocialsList.tsx [hunk 1]
```diff
@@ -32,6 +33,11 @@ const socials = [
 		logo: TwitterLogo,
 		text: "Twitter",
 	},
+	{
+		href: "https://github.com/SquiggleTools/boston-ts-website",
```
Comment (astorije) [inline app/components/SocialsList.tsx:37]: @JoshuaKGoldberg, do we want to link to `https://github.com/SquiggleTools/boston-ts-website` or `https://github.com/SquiggleTools`?

File: app/components/logos/GitHubLogo.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+//adding new file for the githublogo
```
Comment (astorije) [inline app/components/logos/GitHubLogo.tsx:1]: You won't need this comment here :)

```suggestion
```

**Comment by astorije** [comment]
This is great, thank you for the timely contribution! Where did you obtain the SVG code for the GitHub logo?

File: app/components/SocialsList.tsx (hunk 1 above)
Comment (JoshuaKGoldberg) [reply app/components/SocialsList.tsx:37]: I think the `boston-ts-website`. https://github.com/SquiggleTools will also have SquiggleConf and likely more stuff too.

**Comment by astorije** [comment]
Hey @DannyGarciaDEV, are you planning on addressing the comments on this PR? 🙏

**Comment by DannyGarciaDEV** [comment]
Sorry I just saw this messages! I got it from here but i had to do some changes https://iconmonstr.com/github-1-svg/

**Comment by astorije** [comment]
No worries, thanks! I committed my suggestion so make sure to run `git pull` before making any new commits.

Re: logo, what changes have you had to make? Could we instead use the SVG from https://github.com/logos as it's the official one?

File: app/components/SocialsList.tsx [hunk 3]
```diff
@@ -5,6 +5,7 @@ import { DiscordLogo } from "./logos/DiscordLogo";
 import { LinkedInLogo } from "./logos/LinkedInLogo";
 import { MastodonLogo } from "./logos/MastodonLogo";
 import { TwitterLogo } from "./logos/TwitterLogo";
+import { GitHubLogo } from "./logos/GitHubLogo";
```
Comment (astorije) [inline app/components/SocialsList.tsx:8]: You need to sort these imports as such:

```suggestion
import { GitHubLogo } from "./logos/GitHubLogo";
import { TwitterLogo } from "./logos/TwitterLogo";
```

**Comment by astorije** [comment]
Hey @DannyGarciaDEV, quick follow-up, would you like me to take over this PR? No pressure of course :)

File: app/components/logos/GitHubLogo.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { SocialLogo, SocialLogoProps } from "./SocialLogo";
+
+export function GitHubLogo(props: SocialLogoProps) {
+	return (
+		<SocialLogo height="24" viewBox="0 10 100 81" width="24" {...props}>
+			<path
+				d="M48.854 0C21.839 0 0 22 0 49.217c0 21.756 13.993 40.172 33.405 46.69 2.427.49 3.316-1.059 3.316-2.362 0-1.141-.08-5.052-.08-9.127-13.59 2.934-16.42-5.867-16.42-5.867-2.184-5.704-5.42-7.17-5.42-7.17-4.448-3.015.324-3.015.324-3.015 4.934.326 7.523 5.052 7.523 5.052 4.367 7.496 11.404 5.378 14.235 4.074.404-3.178 1.699-5.378 3.074-6.6-10.839-1.141-22.243-5.378-22.243-24.283 0-5.378 1.94-9.778 5.014-13.2-.485-1.222-2.184-6.275.486-13.038 0 0 4.125-1.304 13.426 5.052a46.97 46.97 0 0 1 12.214-1.63c4.125 0 8.33.571 12.213 1.63 9.302-6.356 13.427-5.052 13.427-5.052 2.67 6.763.97 11.816.485 13.038 3.155 3.422 5.015 7.822 5.015 13.2 0 18.905-11.404 23.06-22.324 24.283 1.78 1.548 3.316 4.481 3.316 9.126 0 6.6-.08 11.897-.08 13.526 0 1.304.89 2.853 3.316 2.364 19.412-6.52 33.405-24.935 33.405-46.691C97.707 22 75.788 0 48.854 0z"
```
Comment (astorije) [inline app/components/logos/GitHubLogo.tsx:7]: Confirmed that this is the same SVG as available at https://github.com/logos 👌

File: app/components/SocialsList.tsx [hunk 5]
```diff
@@ -4,6 +4,7 @@ import { BlueskyLogo } from "./logos/BlueskyLogo";
 import { DiscordLogo } from "./logos/DiscordLogo";
 import { LinkedInLogo } from "./logos/LinkedInLogo";
 import { MastodonLogo } from "./logos/MastodonLogo";
+import { GitHubLogo } from "./logos/GitHubLogo";
```
Comment (astorije) [inline app/components/SocialsList.tsx:7]:
```suggestion
import { GitHubLogo } from "./logos/GitHubLogo";
import { LinkedInLogo } from "./logos/LinkedInLogo";
import { MastodonLogo } from "./logos/MastodonLogo";
```

**Review by astorije — APPROVED** [review]
Amazing, thank you @DannyGarciaDEV!

**Comment by JoshuaKGoldberg** [comment]
@all-contributors please add @dannygarciadev for code.

> 🤖 Beep boop! This comment was added automatically by [all-contributors-auto-action](https://github.com/marketplace/actions/all-contributors-auto-action).
> Not all contributions can be detected from Git & GitHub alone. Please comment any missing contribution types this bot missed.
> ...and of course, thank you for contributing! 💙

**Pull request merged**